src/main/java/com/artipie/RepositoriesFromStorage.java

```diff
@@ -35,33 +35,33 @@
     private static LoadingCache<KeyAndStorage, Single<StorageAliases>> aliases;
 
     static {
-        final CacheLoader<KeyAndStorage, Single<String>> ldrconfigs = new CacheLoader<>() {
-            @Override
-            public Single<String> load(final KeyAndStorage pair) {
-                return Single.fromFuture(
-                    new ConfigFile(pair.key)
-                        .valueFrom(pair.storage)
-                        .thenApply(PublisherAs::new)
-                        .thenCompose(PublisherAs::asciiString)
-                        .toCompletableFuture()
-                );
-            }
-        };
-        final CacheLoader<KeyAndStorage, Single<StorageAliases>> ldralias = new CacheLoader<>() {
-            @Override
-            public Single<StorageAliases> load(final KeyAndStorage pair) {
-                return Single.fromFuture(
-                    StorageAliases.find(pair.storage, pair.key)
-                );
-            }
-        };
-        final int timeout = new ArtipieProperties().configCacheTimeout();
+        System.setProperty(
+            ArtipieProperties.CONFIG_TIMEOUT,
+            new ArtipieProperties().configCacheTimeout()
+        );
+        final int timeout = Integer.getInteger(ArtipieProperties.CONFIG_TIMEOUT, 2 * 60 * 1000);
         RepositoriesFromStorage.configs = CacheBuilder.newBuilder()
             .expireAfterWrite(timeout, TimeUnit.MILLISECONDS)
-            .softValues().build(ldrconfigs);
+            .softValues()
+            .build(
```

Comment:
I mean can we use something like `build(config -> config.configContent())` here? Or `CacheLoader` is not a `FunctionalInterface`?

Reply:
@g4s8 got it. `CacheLoader` is an abstract class.